packages/cactus-plugin-ledger-connector-polkadot/src/main/typescript/plugin-ledger-connector-polkadot.ts

```diff
@@ -0,0 +1,504 @@
+}
+
+interface RunTransactionResponse {
+  success: boolean;
+  hash: Hash | undefined;
+}
+
+export interface DeployContractInkBytecodeRequest {
+  wasm: Uint8Array;
+  abi: AnyJson;
+  endowment: number;
+  gasLimit: number;
+  params: Array<unknown>;
+}
+
+interface DeployContractInkBytecodeResponse {
+  success: boolean;
+  contractCode: CodePromise;
+  contractAbi: Abi;
+}
+
+export interface ReadStorageRequest {
+  transferSubmittable: SubmittableExtrinsic<"promise">;
+}
+
+interface ReadStorageResponse {
+  success: boolean;
+  hash: Hash | undefined;
+}
+
+export interface WriteStorageRequest {
+  transferSubmittable: SubmittableExtrinsic<"promise">;
+}
+
+interface WriteStorageResponse {
+  success: boolean;
+  hash: Hash | undefined;
+}
+
+export interface ResponseContainer {
+  response_data: SignerOptions;
+  succeeded: boolean;
+  message: string;
+  error: unknown;
+}
+
+export class PluginLedgerConnectorPolkadot
+  implements
+    IPluginLedgerConnector<
+      DeployContractInkBytecodeRequest,
+      DeployContractInkBytecodeResponse,
+      RunTransactionRequest,
+      RunTransactionResponse
+    >,
+    ICactusPlugin,
+    IPluginWebService {
+  public static readonly CLASS_NAME = "PluginLedgerConnectorPolkadot";
+  private readonly instanceId: string;
+  private readonly log: Logger;
+  public wsProvider: WsProvider | undefined;
+  public api: ApiPromise | undefined;
+  private contractAbi?: Abi | null;
+  private contractCode?: CodePromise | null;
+  public prometheusExporter: PrometheusExporter;
+  private endpoints: IWebServiceEndpoint[] | undefined;
+  private pluginRegistry: PluginRegistry | undefined;
+
+  public getOpenApiSpec(): unknown {
+    return OAS;
+  }
+
+  public get className(): string {
+    return PluginLedgerConnectorPolkadot.CLASS_NAME;
+  }
+
+  constructor(public readonly opts: IPluginLedgerConnectorPolkadotOptions) {
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(opts, `${fnTag} arg options`);
+    if (typeof opts.logLevel !== "undefined") {
+      Checks.truthy(opts.logLevel, `${fnTag} options.logLevelDesc`);
+    }
+    Checks.truthy(opts.wsProviderUrl, `${fnTag} options.wsProviderUrl`);
+    Checks.truthy(opts.instanceId, `${fnTag} options.instanceId`);
+
+    this.prometheusExporter =
+      opts.prometheusExporter ||
+      new PrometheusExporter({ pollingIntervalInMin: 1 });
+    Checks.truthy(
+      this.prometheusExporter,
+      `${fnTag} options.prometheusExporter`,
+    );
+
+    const level = this.opts.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+
+    this.pluginRegistry = opts.pluginRegistry;
+    this.instanceId = opts.instanceId;
+    this.setProvider(opts.wsProviderUrl);
+    this.prometheusExporter.startMetricsCollection();
+  }
+
+  public setProvider(wsProviderUrl: string): void {
+    try {
+      this.wsProvider = new WsProvider(wsProviderUrl);
```

Comment:
Does this even try to connect to the `wsProviderUrl`?
If yes, is there a way to avoid autoconnect by passing some other argument(s) to the `WsProvider` constructor or some other way? (If this is enabled, this will be helpful in the DAST scan, so that we can pass a dummy `wsProviderUrl` and don't require to spin up an actual `all-in-one` image for Polkadot)
@petermetz will we be adding this platform as well to the DAST scans?

Reply:
Yes it automatically connects (behind the scenes): https://polkadot.js.org/docs/api/start/create/
and yes.